File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        Collections.sort(intervals, (a, b) -> Integer.compare(a.start, b.start));
        if (intervals.isEmpty()) return 0;
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        int max = 1;
        minHeap.offer(intervals.get(0).end);
        for (int i = 1; i < intervals.size(); i++) {
            while (!minHeap.isEmpty() && intervals.get(i).start >= minHeap.peek()) {
                minHeap.poll();
            }
            minHeap.offer(intervals.get(i).end);
            max = Math.max(max, minHeap.size());
        }
        return max;
    }
}

File: Data Structures & Algorithms/is-subsequence/submission-2.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(t) < len(s): return False
        if len(s) == 0: return True
        ps, pt = 0,0

        while pt < len(t):
            if ps < len(s) and t[pt] == s[ps]:
                ps += 1
            pt += 1

        return ps == len(s)